stdlib/oauth2/src/main/ballerina/src/oauth2/outbound_oauth2_provider.bal

```diff
@@ -91,12 +90,9 @@ public type OutboundOAuth2Provider object {
     public function generateToken() returns @tainted (string|auth:Error) {
         GrantTypeConfig? oauth2ProviderConfig = self.oauth2ProviderConfig;
         if (oauth2ProviderConfig is ()) {
-            runtime:AuthenticationContext? authContext = runtime:getInvocationContext()?.authenticationContext;
-            if (authContext is runtime:AuthenticationContext) {
-                string? authToken = authContext?.authToken;
-                if (authToken is string) {
-                    return authToken;
-                }
+            string? authToken = auth:getInvocationContext()?.token;
+            if (authToken is string) {
+                return authToken;
             }
             return prepareAuthError("Failed to generate OAuth2 token since OAuth2 provider config is not defined and auth token is not defined in the authentication context at invocation context.");
```

Comment:
```suggestion
            return prepareAuthError("Failed to generate OAuth2 token since OAuth2 provider config is not defined and  OAuth2 token is not defined at auth:InvocationContext.");
```